zhaquirks/tuya/__init__.py

```diff
@@ -94,18 +111,12 @@ async def write_attributes(self, attributes, manufacturer=None):
         records = self._write_attr_records(attributes)
 
         for record in records:
-            # serialized in little-endian
-            data = list(record.value.value.serialize())
-            # we want big-endian, with length prepended
-            data.append(len(data))
-            data.reverse()
-
             cmd_payload = TuyaManufCluster.Command()
             cmd_payload.status = 0
             cmd_payload.tsn = self.endpoint.device.application.get_sequence()
             cmd_payload.command_id = record.attrid
             cmd_payload.function = 0
-            cmd_payload.data = data
+            cmd_payload.data = Data.from_value(record.value.value)
 
             await super().command(
```

Comment:
Anyone knows if Tuya replies with any Success/Failure statuses when setting an attribute?

Reply:
Do you need loog from one TRV then setting some attribute ?
If yes please provide log/debug settings for getting the right logs and what attributes i should changing.

Reply:
The tuya siren I have does not reply to the `set_data` command. If I use `expect_reply=True` I get a timeout. What I *do* get is a `set_data` command from the device with the updated value (which is why I do not call `_update_attribute` here).